flink-libraries/flink-table/src/main/scala/org/apache/flink/table/descriptors/BatchTableDescriptor.scala

```diff
@@ -0,0 +1,126 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.descriptors
+
+import org.apache.flink.table.api.{BatchTableEnvironment, ValidationException}
+import org.apache.flink.table.factories.{BatchTableSinkFactory, BatchTableSourceFactory, TableFactoryService}
+
+/**
+  * Descriptor for specifying a table source and/or sink in a batch environment.
+  */
+class BatchTableDescriptor(
+    private val tableEnv: BatchTableEnvironment,
+    private val connectorDescriptor: ConnectorDescriptor)
+  extends TableDescriptor
+  with SchematicDescriptor
+  with RegistrableDescriptor {
+
+  private var formatDescriptor: Option[FormatDescriptor] = None
+  private var schemaDescriptor: Option[Schema] = None
+
+  /**
+    * Internal method for properties conversion.
+    */
+  override private[flink] def addProperties(properties: DescriptorProperties): Unit = {
```

Comment:
in many places (for example all `addProperties` methods) you are ordering methods very weirdly. Rule of thumb should be pubic methods before private. Longer story: https://stackoverflow.com/a/1760877/8149051

Many times in this code review I had to jump up & down.

Reply:
Depends on the language you are using. This method is public in Java ;)
But I will move it down.